File: app/src/main/java/io/github/freewebmovement/igniter/servers/contract/ServerListContract.java
package io.github.freewebmovement.igniter.servers.contract;

import android.content.Context;
import android.net.Uri;

import java.util.List;

import io.github.freewebmovement.igniter.persistence.TrojanConfig;
import io.github.freewebmovement.igniter.common.mvp.BasePresenter;
import io.github.freewebmovement.igniter.common.mvp.BaseView;

public interface ServerListContract {
    interface Presenter extends BasePresenter {
        void addServerConfig(String trojanUrl);
        void handleServerSelection(TrojanConfig config);
        void deleteServerConfig(TrojanConfig config, int pos);
        void gotoScanQRCode();
        void displayImportFileDescription();
        void hideImportFileDescription();
        void importConfigFromFile();
        void parseConfigsInFileStream(Context context, Uri fileUri);
    }

    interface View extends BaseView<Presenter> {
        void showAddTrojanConfigSuccess();
        void showQRCodeScanError(String scanContent);
        void selectServerConfig(TrojanConfig config);
        void showServerConfigList(List<TrojanConfig> configs);
        void removeServerConfig(TrojanConfig config, int pos);
        void gotoScanQRCode();
        void showImportFileDescription();
        void dismissImportFileDescription();
        void openFileChooser();
    }
}

File: app/src/main/java/io/github/freewebmovement/igniter/activities/servers/contract/ServerListContract.kt
package io.github.freewebmovement.igniter.activities.servers.contract

import android.content.Context
import android.net.Uri
import io.github.freewebmovement.igniter.common.mvp.BasePresenter
import io.github.freewebmovement.igniter.common.mvp.BaseView
import io.github.freewebmovement.igniter.persistence.TrojanConfig

interface ServerListContract {
    interface Presenter : BasePresenter {
        fun handleServerSelection(config: TrojanConfig)
        fun deleteServerConfig(config: TrojanConfig, pos: Int)
        fun displayImportFileDescription()
        fun hideImportFileDescription()
        fun importConfigFromFile()
        fun parseConfigsInFileStream(context: Context, fileUri: Uri)
    }

    interface View : BaseView<Presenter> {
        fun showAddTrojanConfigSuccess()
        fun selectServerConfig(config: TrojanConfig)
        fun showServerConfigList(configs: List<TrojanConfig>)
        fun removeServerConfig(config: TrojanConfig, pos: Int)
        fun showImportFileDescription()
        fun dismissImportFileDescription()
        fun openFileChooser()
    }
}
